### ISSUE TITLE
Time stamp issues with binance API 

### ISSUE THREAD

**Issue by blinkcloud** [jaggedsoft/node-binance-api#68]
"code":-1021,"msg":"Timestamp for this request was 1000ms ahead of the server\'s time."

**Comment by patkaerndel** [comment]
1021

**Comment by jaggedsoft** [comment]
This happens when your system time is out of sync

**Comment by jaggedsoft** [comment]
Still having problems? Try adding `recvWindow` to your options:

```js
binance.options({
	'APIKEY': 'xxx',
	'APISECRET': 'xxx',
	'recvWindow': 60000
});
```

**Comment by jaggedsoft** [comment]
Please tell me if the problem persists
If it does, a solution that I can think of is to synchronize with the servers time when the program is first started

**Comment by usama33** [comment]
Yes, the problem still exists.. I tried different ranges for recWindown but it didn't work.

**Comment by jaggedsoft** [comment]
1. You can now use the `useServerTime` option to synchronize with the servers time at startup
2. You can now pass a callback function to options() which runs after the synchronization is complete

```js
binance.options({
	APIKEY: '<key>',
	APISECRET: '<secret>',
	useServerTime: true
}, main);

function main() {
	binance.balance((error, balances) => {
		if ( error ) {
			console.error(error);
			return;
		}
		console.log("balances()", balances);
		console.log("BNB balance: ", balances.BNB.available);
	});
}
```

**Comment by jaggedsoft** [comment]
Here is another way to do it, which will be available soon in `v0.4.6`

```js
binance.useServerTime(function() {
	binance.balance((error, balances) => {
		if ( error ) return console.error(error);
		console.log("balances()", balances);
		console.log("BNB balance: ", balances.BNB.available);
	});
});
```
